code/optimize/src/main/java/com/adobe/marketing/mobile/optimize/Optimize.java

```diff
@@ -118,6 +118,136 @@ public static void updatePropositions(
         MobileCore.dispatchEvent(event);
     }
 
+    /**
+     * This API dispatches an Event for the Edge network extension to fetch decision propositions,
+     * for the provided decision scopes list, from the decisioning services enabled in the
+     * Experience Edge network.
+     *
+     * <p>The returned decision propositions are cached in-memory in the Optimize SDK extension and
+     * can be retrieved using {@link #getPropositions(List, AdobeCallback)} API.
+     *
+     * @param decisionScopes {@code List<DecisionScope>} containing scopes for which offers need to
+     *     be updated.
+     * @param xdm {@code Map<String, Object>} containing additional XDM-formatted data to be sent in
+     *     the personalization query request.
+     * @param data {@code Map<String, Object>} containing additional free-form data to be sent in
+     *     the personalization query request.
+     * @param callback {@code AdobeCallback<Map<DecisionScope, OptimizeProposition>>} which will be
+     *     invoked when decision propositions are received from the Edge network.
+     */
+    public static void updatePropositions(
```

Comment:
please check if we still need to keep the existing api, if we decide to keep the callback optional (nullable) will it ensure backward compatibility or not?

Reply:
I have marked the callback as nullable now, we'll still need to keep the existing API signature to maintain backward compatibility